### PR TITLE
RD-6159 Disable Set Site if there are no sites

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Jira ticket: RD-6159

## Screenshots / Videos
Before:
Screenshot 2023-01-27 at 13 28 33

After:
Screenshot 2023-01-27 at 13 28 02

Demo video:

https://user-images.githubusercontent.com/17104212/215075482-cfdbf962-24e5-444b-a3c4-bb71c72ede0a.mov

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

`deployment_actions_spec` 17-February-2023 -  ✅ passed:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4529/pipeline

`deployment_actions_spec` 16-February-2023 -  ✅ passed:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4525/pipeline/

`deployment_actions_spec` 06-February-2023:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4467/pipeline

`deployment_actions_spec` after fixing the implementation:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4424/pipeline

`deployment_actions_spec`:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4422/pipeline

 

## Documentation
There is nothing to add: https://docs.cloudify.co/latest/working_with/console/widgets/deploymentactionbuttons/

### CODE REVIEW

File: app/widgets/common/deployments/useSitesExist.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        toolbox
+            .getManager()
+            .doGet('/sites?_size=1')
+            .then((response: any) => {
+                const sites = response.items;
+                if (sites.length > 0) {
+                    setSitesExist(true);
+                } else {
+                    setSitesExist(false);
+                }
```
Comment (kubama) [inline app/widgets/common/deployments/useSitesExist.ts:16]:
```suggestion
                setSitesExist(sites.length > 0);
```

Comment (leon-good-life) [reply app/widgets/common/deployments/useSitesExist.ts]: [9b3a193](https://github.com/cloudify-cosmo/cloudify-stage/pull/2415/commits/9b3a193ac4e13652041d3cfbc9552add983d91c6)

**Review by kubama — APPROVED** [review]

**Comment by leon-good-life** [comment]
I fixed the test (reloading the menu component). It passed locally. I run it on jenkins:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4524/pipeline

**Comment by leon-good-life** [comment]
I fixed the test (reloading the menu component). It passed locally. I run it on jenkins - ✅ passed:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4525/pipeline/

File: test/cypress/integration/widgets/deployment_actions_spec.ts [hunk 2]
```diff
@@ -57,9 +57,20 @@ describe('Deployment Action Buttons widget', () => {
...
 
+        it('should disable set site if there are no sites', () => {
+            cy.deleteSites();
+
+            cy.contains('button', 'Deployment actions').should('not.have.attr', 'disabled');
+            cy.clickButton('Deployment actions');
+
+            cy.get('.popupMenu > .menu').contains('Set Site').should('have.class', 'disabled');
+        });
+
```
Comment (kubama) [inline test/cypress/integration/widgets/deployment_actions_spec.ts:68]: Is there any reason for moving this test case? If not, I would leave it in its original position to keep the diff small.

Comment (leon-good-life) [reply test/cypress/integration/widgets/deployment_actions_spec.ts]: I moved it back: [ab392c0](https://github.com/cloudify-cosmo/cloudify-stage/pull/2415/commits/ab392c0b39091ba3ad815bd49fc6e7aa61a01166)

And I run the test again:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4529/pipeline
✅ passed

**Review by kubama — APPROVED** [review]

**Pull request merged**